packages/components/src/confirm-dialog/component.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import { forwardRef, useEffect, useState } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import type { Ref, SyntheticEvent, KeyboardEvent } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Modal from '../modal';
+import type { OwnProps } from './types';
+import {
+	useContextSystem,
+	contextConnect,
+	WordPressComponentProps,
+} from '../ui/context';
+import { Flex } from '../flex';
+import Button from '../button';
+import * as styles from './styles';
+import { useCx } from '../utils/hooks/use-cx';
+
+function ConfirmDialog(
+	props: WordPressComponentProps< OwnProps, 'div', false >,
+	forwardedRef: Ref< any >
+) {
+	const {
+		isOpen,
+		message,
+		onConfirm,
+		onCancel = () => {},
```

Comment:
This default assignment would create a new function every time the component gets rendered.

An alternative could be to not assign any default value, and then call this function with optional chaining, e.g. `onCancel?.()`

Reply:
Sounds good 👍🏻

Reply:
I'm totally fine changing to use the optional chaining at the call place, but I'm curious how much an empty function definition could affect the rendering performance -- couldn't this be considered a premature optimization?

Reply:
You have a point, for sure — this default assignment won't really affect page performance.

To me, it's more about applying the same "conventions" and good practices that we'd apply to other components. Does it make sense? Do you think the one I'm suggesting is not a good practice? I'm always open to such conversations :)

Reply:
> To me, it's more about applying the same "conventions" and good practices that we'd apply to other components. Does it make sense? Do you think the one I'm suggesting is not a good practice?

What you say about conventions and best practices makes sense. As to what I think, I don't have a strong opinion here, but from a clarity point-of-view, what you suggest makes sense I think. One could argue that calling it with optional chaining better communicates it's optional than just assigning a noop function 👍🏻